aderyn_core/src/detect/high/incorrect_shift_order.rs

```diff
@@ -0,0 +1,90 @@
+use std::collections::BTreeMap;
+use std::error::Error;
+
+use crate::ast::{NodeID, YulExpression};
+
+use crate::capture;
+use crate::detect::detector::IssueDetectorNamePool;
+use crate::{
+    context::workspace_context::WorkspaceContext,
+    detect::detector::{IssueDetector, IssueSeverity},
+};
+use eyre::Result;
+
+#[derive(Default)]
+pub struct IncorrectShiftOrderDetector {
+    // Keys are: [0] source file name, [1] line number, [2] character location of node.
+    // Do not add items manually, use `capture!` to add nodes to this BTreeMap.
+    found_instances: BTreeMap<(String, usize, String), NodeID>,
+}
+
+impl IssueDetector for IncorrectShiftOrderDetector {
+    fn detect(&mut self, context: &WorkspaceContext) -> Result<bool, Box<dyn Error>> {
+        let yul_function_calls = context.yul_function_calls();
+        for yul_function_call in yul_function_calls {
+            if yul_function_call.function_name.name == "shl"
+                || yul_function_call.function_name.name == "shr"
+            {
+                if let YulExpression::YulLiteral(_) = &yul_function_call.arguments[1] {
+                    capture!(self, context, yul_function_call);
+                }
```

Comment:
Yul function calls don't have IDs, how are we able to capture these ??  (Confused) 😕

Reply:
(I'm trying to adjust the big refactor PR for Yul stuff)

Reply:
Ahh I got it - the capture! macro was changed to insert `0` as the id if it's none !